crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentCompletionService.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.justice.laa.crime.meansassessment.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.meansassessment.dto.MeansAssessmentDTO;
+import uk.gov.justice.laa.crime.meansassessment.dto.MeansAssessmentRequestDTO;
+import uk.gov.justice.laa.crime.meansassessment.dto.maatcourtdata.DateCompletionRequestDTO;
+import uk.gov.justice.laa.crime.meansassessment.dto.maatcourtdata.FinancialAssessmentDTO;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.*;
+
+import java.time.LocalDateTime;
+
+import static java.util.Optional.ofNullable;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AssessmentCompletionService {
+
+    private final MaatCourtDataService maatCourtDataService;
+
+    public void execute(MeansAssessmentDTO assessment, String laaTransactionId) {
+        boolean isUpdateRequired = false;
+        MeansAssessmentRequestDTO meansAssessmentRequest = assessment.getMeansAssessment();
+        CurrentStatus assessmentStatus = meansAssessmentRequest.getAssessmentStatus();
+        AssessmentType type = meansAssessmentRequest.getAssessmentType();
+        if (CurrentStatus.COMPLETE.equals(assessmentStatus)) {
+            if (AssessmentType.INIT.equals(type)) {
+                isUpdateRequired = isInitAssessmentComplete(assessment);
+            } else {
+                isUpdateRequired = isFullUpdateRequired(assessment, laaTransactionId);
+            }
+        }
+
+        if (isUpdateRequired) {
+            updateApplicationCompletionDate(assessment, laaTransactionId);
+        }
+    }
+
+    void updateApplicationCompletionDate(MeansAssessmentDTO assessment, String laaTransactionId) {
+        assessment.setDateCompleted(LocalDateTime.now());
+        DateCompletionRequestDTO dateCompletionRequestDTO = DateCompletionRequestDTO
+                .builder()
+                .repId(assessment.getMeansAssessment().getRepId())
+                .assessmentDateCompleted(assessment.getDateCompleted())
+                .build();
+        maatCourtDataService.updateCompletionDate(dateCompletionRequestDTO, laaTransactionId);
+    }
+
+    boolean isFullUpdateRequired(MeansAssessmentDTO assessment, String laaTransactionId) {
+        Integer financialAssessmentId = ofNullable(
```

Comment:
Do we need a ofNullable check here? Is this not redundant?

Reply:
Yes, it is, not sure why I thought I needed it.